### PR TITLE
Add raw images information to augment methods in VectorizedBaseImageAugmentationLayer

### LINKED ISSUE #1590: No raw images information for bounding boxes, segmentation masks and keypoints augmentation in `VectorizedBaseImageAugmentationLayer`
Many KPLs require raw images information to augment bounding boxes with  bounding box converter.

Original BaseImageAugmentationLayer implements this behavior:

https://github.com/keras-team/keras-cv/blob/61c231a01cba2b5ee3709d3351fb05b023274108/keras_cv/layers/preprocessing/base_image_augmentation_layer.py#L406-L409

https://github.com/keras-team/keras-cv/blob/61c231a01cba2b5ee3709d3351fb05b023274108/keras_cv/layers/preprocessing/base_image_augmentation_layer.py#L451-L461

But VectorizedBaseImageAugmentationLayer don't:

https://github.com/keras-team/keras-cv/blob/61c231a01cba2b5ee3709d3351fb05b023274108/keras_cv/layers/preprocessing/vectorized_base_image_augmentation_layer.py#L264-L269

https://github.com/keras-team/keras-cv/blob/61c231a01cba2b5ee3709d3351fb05b023274108/keras_cv/layers/preprocessing/vectorized_base_image_augmentation_layer.py#L311-L319

For example, RandomCrop requires `raw_image` for `bounding_box.convert_format`

https://github.com/keras-team/keras-cv/blob/61c231a01cba2b5ee3709d3351fb05b023274108/keras_cv/layers/preprocessing/random_crop.py#L94-L109

This issue can be resolved by either replacing images with raw images when using augment methods or by adding a new argument called `raw_images` to these methods.
However, it is unclear whether KPL will require processed images or raw images for augmentation in the future.

I can open a PR once it becomes more developed and concrete.

### PR DESCRIPTION
**Pull request by james77777778**
# What does this PR do?

Fixes #1590 

Moreover, this PR adds `keras.utils.register_keras_serializable` to VectorizedBaseImageAugmentationLayer, as well as fixes for some missing layers in the `__init__.py`.

The unit test has been updated to ensure that `raw_images` is present.

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [X] Did you read the [contributor guideline](https://github.com/keras-team/keras-cv/blob/master/.github/CONTRIBUTING.md),
      Pull Request section?
- [X] Was this discussed/approved via a Github issue? Please add a link
      to it if that's the case.
- [X] Did you write any new necessary tests?
- [ ] If this adds a new model, can you run a few training steps on TPU in Colab to ensure that no XLA incompatible OP are used?

## Who can review?
@LukeWood @ianstenbit

### CODE REVIEW

**Review by LukeWood — APPROVED** [review]

**Comment by LukeWood** [comment]
Thanks @james77777778
